ankihub/gui/deck_browser.py

```diff
@@ -0,0 +1,105 @@
+"""Modifies the Anki deck browser (aqt.deck_browser)."""
+
+import aqt
+from anki.decks import DeckId
+from anki.hooks import wrap
+from aqt import QMenu, gui_hooks, qconnect
+
+from .. import LOGGER
+from ..gui.subdeck_due_date_dialog import DatePickerDialog
+from ..main.block_exam_subdecks import move_subdeck_to_main_deck
+from ..main.deck_unsubscribtion import unsubscribe_from_deck_and_uninstall
+from ..settings import (
+    BlockExamSubdeckConfig,
+    config,
+)
+from .utils import ask_user
+
+
+def setup() -> None:
+    """Ask the user if they want to unsubscribe from the AnkiHub deck when they delete the associated Anki deck."""
+
+    def _after_anki_deck_deleted(did: DeckId) -> None:
+        deck_ankihub_id = config.get_deck_uuid_by_did(did)
+        if not deck_ankihub_id:
+            return
+        deck_name = config.deck_config(deck_ankihub_id).name
+        if ask_user(
+            text="You've deleted the Anki deck linked to the<br>"
+            f"<b>{deck_name}</b> AnkiHub deck.<br><br>"
+            "Do you also want to unsubscribe from this AnkiHub deck to avoid receiving future updates?<br><br>"
+            "For more info, check out "
+            "<a href='https://community.ankihub.net/t/how-are-anki-decks-related-to-ankihub-decks/4811/1'>"
+            "this topic on our forum</a>.",
+            title="Unsubscribe from AnkiHub Deck?",
+            parent=aqt.mw,
+        ):
+            unsubscribe_from_deck_and_uninstall(deck_ankihub_id)
+
+    aqt.mw.deckBrowser._delete = wrap(  # type: ignore
+        old=aqt.mw.deckBrowser._delete,
+        new=_after_anki_deck_deleted,
+    )
+
+
+def _open_dialog_date_picker_for_subdeck(subdeck_config: BlockExamSubdeckConfig) -> None:
+    if not subdeck_config:
+        LOGGER.warning("Subdeck with ID %s not found in configuration.", subdeck_config.subdeck_id)
+        return
```

Comment:
⛏️ We can remove this check, the function will only be called if the the subdeck config exists.

```suggestion
```